src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -848,52 +848,6 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
             return sum;
         }
 
-        internal static Dictionary<K, ImmutableArray<T>> ToDictionary<K, T>(this ImmutableArray<T> items, Func<T, K> keySelector)
-            where K : notnull
-            where T : notnull
-        {
-            if (items.Length == 1)
-            {
-                T value = items[0];
-                return new Dictionary<K, ImmutableArray<T>>(1)
-                {
-                    {  keySelector(value), ImmutableArray.Create(value) },
-                };
-            }
-
-            if (items.Length == 0)
-            {
-                return new Dictionary<K, ImmutableArray<T>>();
-            }
-
-            // bucketize
-            // prevent reallocation. it may not have 'count' entries, but it won't have more. 
-            //
-            // We store a mapping from keys to either a single item (very common in practice as this is used from
-            // callers that maps names to symbols with that name, and most names are unique), or an array builder of items.
-
-            var accumulator = PooledDictionary<K, object>.GetInstance();
-            foreach (var item in items)
-            {
-                var key = keySelector(item);
-                AddToMultiValueDictionaryBuilder(accumulator, key, item);
-            }
-
-            var dictionary = new Dictionary<K, ImmutableArray<T>>(accumulator.Count);
-
-            // freeze
-            foreach (var pair in accumulator)
-            {
-                dictionary.Add(pair.Key, pair.Value is ArrayBuilder<T> arrayBuilder
-                    ? arrayBuilder.ToImmutableAndFree()
-                    : ImmutableArray.Create((T)pair.Value));
-            }
-
-            accumulator.Free();
-
-            return dictionary;
-        }
-
         internal static void AddToMultiValueDictionaryBuilder<K, T>(Dictionary<K, object> accumulator, K key, T item)
```

Comment:
this member stays.  it is still used in several places (including VB and hte IDE layer).